RHIDP-5592:Wrong definition of Service Monitor for an RHDH instance by the Operator (#873)
redhat-developer/red-hat-developers-documentation-rhdh: modules/observe/proc-admin-enabling-metrics-ocp-operator.adoc

apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:

>   name: <service-monitor-name> <1>

Reviewer: Since we're filling in the rest of the values via the attributes, we can fill this value in directly as well with `developer-hub-service-monitor` for easy copying and pasting

>   namespace: <namespace> <2>

Reviewer 2: Use the `{my-product-namespace}` attribute. We just fixed all the codebase in that direction. See https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/858

  labels:
    app.kubernetes.io/instance: {my-product-cr-name} <3>
    app.kubernetes.io/name: {product-custom-resource-type}
spec:
  namespaceSelector:
    matchNames:
      - {my-product-namespace} <2>
  selector:
    matchLabels:

>       app.kubernetes.io/instance: {my-product-cr-name} <3>

Reviewer: This value differs from the name of the `ServiceMonitor` resource (`metadata.labels.app.kubernetes.io/instance`) we are creating. It should be the name of the RHDH deployment. We're trying to select the RHDH deployment to target the monitoring to.

      app.kubernetes.io/name: backstage
  endpoints:
  - port: http-metrics
    path: '/metrics'
----
<1> The name of your ServiceMonitor resource, for example, `developer-hub-service-monitor`.

> <2> The namespace where your ServiceMonitor lives and where your {product-very-short} instance is installed, for example, `rhdh-operator`.

Reviewer 2: Use the `{my-product-namespace}` attribute. We just fixed all the codebase in that direction. See https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/858

> <3> The name of the {ocp-short} project where your {product} instance is running.

Reviewer: This description is a bit confusing, it is the name of the RHDH deployment (something like `developer-hub`) - I'm not sure what the standard value is used in the docs.

apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:

>   name: developer-hub-service-monitor <1>

Reviewer 3: ```suggestion
  name: <developer-hub-service-monitor-name> <1>
```
to align with the OCP docs and Jessica's feedback, we want to format these as variables (`<variable-format>`) and then put the example in the definitions (`example-format`)

>   namespace: {my-product-namespace} <2>

Reviewer 3: ```suggestion
  namespace: <rhdh-namespace-name> <2>
```
if I'm not mistaken, a customer can name this whatever they want, which is why we use variables and examples. so we'll use the attribute as the example in the definition.

Author: These are great suggestions, @linfraze. This way, the variables and examples (our standard attributes) are also used in the docs! Thank you!

  labels:

>     app.kubernetes.io/instance: {my-product-cr-name} <3>

Reviewer 3: ```suggestion
    app.kubernetes.io/instance: <rhdh-cr-name> <3>
```

    app.kubernetes.io/name: {product-custom-resource-type}
spec:
  namespaceSelector:
    matchNames:

>       - {my-product-namespace} <4>

Reviewer 3: ```suggestion
      - <rhdh-namespace-name> <4>
```

  selector:
    matchLabels:

>       app.kubernetes.io/instance: developer-hub <5>

Reviewer 3: ```suggestion
      app.kubernetes.io/instance: <deployment-name> <5>
```

>       app.kubernetes.io/name: {product-custom-resource-type}

Reviewer 3: ```suggestion
      app.kubernetes.io/name: {rhdh-cr-type} (6)
```

  endpoints:
  - port: http-metrics
    path: '/metrics'
----
<1> The name of your `ServiceMonitor` resource, for example, `developer-hub-service-monitor`.

> <2> The namespace where your `ServiceMonitor` lives.

Reviewer: Changing the tense here might help clarify that we're setting the namespace where this CR will be. 

i.e. `<2> The namespace where your ServiceMonitor will live.`

> <2> The namespace where your `ServiceMonitor` will live.

Reviewer 3: ```suggestion
<2> The namespace where your `ServiceMonitor` will live, for example, `{my-product-namespace}`.
```

> <3> The label name identifying the `ServiceMonitor` CR instance.

Reviewer 3: ```suggestion
<3> The label name identifying the `ServiceMonitor` CR instance, for example, `{my-product-cr-name}`.
```

> <3> The name of the {ocp-short} project where your {product} instance is running.

Reviewer: This is one of the labels that identifies this specific instance of ServiceMonitor and not the name of the project. Maybe describe it as something like: `The label name identifying the ServiceMonitor CR instance`

> <4> The namespace where your {product-very-short} instance is installed.

Reviewer 3: ```suggestion
<4> The namespace where your {product-very-short} instance is installed, for example, `{my-product-namespace}`.
```

> <5> The name of your {product-very-short} deployment. 

Reviewer 3: ```suggestion
<5> The name of your {product-very-short} deployment, for example, `developer-hub`. 
```

> <5> The name of your {product-very-short} deployment. 
> [NOTE]

Reviewer 3: ```suggestion
<5> The name of your {product-very-short} deployment. 
<6> The type of custom resource that you want to use, for example, `{product-custom-resource-type}`.
[NOTE]
```

> [NOTE]
> +
> ====

Reviewer 3: ```suggestion
[NOTE]
====
```

adoc correction to render this as a NOTE admonition

> `spec.selector.matchLabels` configuration must match the labels of your {product-very-short} installation. To confirm the label names, go to Project > Services and verify the labels for `backstage-developer-hub`.

Reviewer 3: ```suggestion
`spec.selector.matchLabels` configuration must match the labels of your {product-very-short} installation. To confirm the label names, go to Project > Services and verify the labels for `backstage-developer-hub`.
```
Since this is a verification step, let's add this as a step in the Verification section and also orient the user and describe the UI in compliance with CCS standards. 

. From the [page where these UI elements live], click *Project* > *Services* to verify the labels for `backstage-developer-hub`.
[NOTE]
===
`spec.selector.matchLabels` configuration must match the labels of your {product-very-short} installation. 
===

apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: <developer-hub-service-monitor-name> <1>

>   namespace: <rhdh-namespace-name> <2>

Reviewer 2: For consistency, use the attribute.

```suggestion
  namespace: <{my-product-namespace}> <2>
```

Author: @themr0c, would it make sense if we used a generic variable and then, use our attributes while describing the example? 
This suggestion is from @linfraze in [this comment](https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/873#discussion_r1932567521).

Reviewer 3: This approach makes sense to me and follows the format that other product docs teams use

  labels:
    app.kubernetes.io/instance: <rhdh-cr-name> <3>
    app.kubernetes.io/name: {product-custom-resource-type}
spec:
  namespaceSelector:
    matchNames:

>       - <rhdh-namespace-name> <4>

Reviewer 2: For consistency, use the attribute.

```suggestion
      - <{my-product-namespace}> <4>
```

Author: I've aligned our variables with the style guide, but I've created a JIRA ticket to change the hyphen to underscores in https://issues.redhat.com/browse/RHIDP-5865.

Reviewer 3: Before our team conversation toady, we were only using attributes in the examples in the definitions and not in the variables in the code sample. 

Based on the RH Supplementary style guide, @pabel-rh is right to convert the hyphens to underscores in the replaceable text. 

We are already making improvements, but before we finalize any decision, we should consult the CCS Accessibility group and other sources to inform the best approach to presenting code samples and commands to the user.

Author: Absolutely, @linfraze. :) 
Would it be okay to approve and merge this PR, and come back to this section if we need to make changes once the best the approach is provided?

  selector:
    matchLabels:
      app.kubernetes.io/instance: <deployment-name> <5>
      app.kubernetes.io/name: <rhdh-cr-type> <6>
  endpoints:
  - port: http-metrics
    path: '/metrics'
----
<1> The name of your `ServiceMonitor` resource, for example, `developer-hub-service-monitor`.
<2> The namespace where your `ServiceMonitor` will live, for example, `{my-product-namespace}`.
<3> The label name identifying the `ServiceMonitor` CR instance, for example, `{my-product-cr-name}`.
<4> The namespace where your {product-very-short} instance is installed, for example, `{my-product-namespace}`.
<5> The name of your {product-very-short} deployment, for example, `developer-hub`.

> <6> The type of custom resource that you want to use, for example, `{product-custom-resource-type}`.

Reviewer: ```suggestion
<6> The name of your {product-very-short} application, for example, `backstage`.
```

. From the *Developer* perspective in the {ocp-short} web console, select the *Observe* view.
. Click the *Metrics* tab to view metrics for {product} pods.

> . From the OpenShift Console in the Developer view, click **Project > Services** and verify the labels for `backstage-developer-hub`.

Reviewer 3: ```suggestion
. From the Developer perspective in the {ocp-short} web console, click **Project > Services** and verify the labels for `backstage-developer-hub`.
```
